File: cs/Core/Platform.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TapstreamMetrics.Sdk
{
    interface Platform
    {
        string LoadUuid();
        HashSet<string> LoadFiredEvents();
        void SaveFiredEvents(HashSet<string> firedEvents);
        string GetResolution();
        string GetManufacturer();
        string GetModel();
        string GetOs();
        string GetLocale();
#if TEST_WINPHONE || WINDOWS_PHONE
        string GetDeviceUniqueId(); 
#else
        string GetAppSpecificHardwareId();
#endif
        string GetAppName();
        string GetPackageName();
        Response Request(string url, string data, string method);
    }
}

File: java/Core/src/com/tapstream/sdk/Platform.java
package com.tapstream.sdk;

import java.util.Set;
import java.util.concurrent.ThreadFactory;

interface Platform {
	public ThreadFactory makeWorkerThreadFactory();

	public String loadUuid();

	public Set<String> loadFiredEvents();

	public void saveFiredEvents(Set<String> firedEvents);

	public String getResolution();

	public String getManufacturer();

	public String getModel();

	public String getOs();

	public String getLocale();

	public String getWifiMac();

	public String getDeviceId();

	public String getAndroidId();

	public String getAppName();

	public String getPackageName();

	public Response request(String url, String data);
}
